### PR TITLE
fix: Recognition dataset merging

### PR DESCRIPTION
**Pull request by charlesmindee**
This PR allows recognition datasets to be merged.
Any feedback is welcome!

### CODE REVIEW

File: doctr/datasets/recognition.py [hunk 1]
```diff
@@ -46,3 +47,19 @@ def __init__(
             if not isinstance(label, str):
                 raise KeyError("Image is not in referenced in label file")
             self.data.append((img_path, label))
+
+    def merge_dataset(self, ds: AbstractDataset) -> None:
+        # Update data with new root for self
+        new_data: List[Tuple[str, str]] = []
+        for img_path, label in self.data:
+            new_data.append((str(Path(self.root).joinpath(img_path)), label))
+        self.data = new_data
```
Comment (fg-mindee) [inline doctr/datasets/recognition.py:56]:
```python
self.data = [(str(Path(self.root).joinpath(img_path)), label) for img,path, label in self.data]
```

list comprehension for the win

File: doctr/datasets/recognition.py [hunk 2]
```diff
@@ -46,3 +47,19 @@ def __init__(
...
+        # Define new root
+        self.root = Path("/")
+        # Merge with ds data
+        for img_path, label in ds.data:
+            self.data.append((str(Path(ds.root).joinpath(img_path)), label))
+
+    def merge_datasets(self, ds_list: List[AbstractDataset]) -> None:
+        for ds in ds_list:
+            self.merge_dataset(ds)
```
Comment (fg-mindee) [inline doctr/datasets/recognition.py:65]: I'm having second thought about whether this method is useful (without it, we only ave one extra line with the "for" loop to add)

**Review by fg-mindee — COMMENTED** [review]
Thanks for the PR! 2 comments where I'd like to have your opinion

File: doctr/datasets/recognition.py (hunk 2 above)
Comment (charlesmindee) [reply doctr/datasets/recognition.py:65]: I agree, as you wish

**Review by fg-mindee — APPROVED** [review]
Thanks!

**Pull request merged**